crates/biome_js_analyze/src/lint/nursery/no_useless_undefined_initialization.rs

```diff
@@ -0,0 +1,135 @@
+    /// let b = undefined, c = 1, d = undefined;
+    /// ```
+    /// ```js,expect_diagnostic
+    /// for (let i = 0; i < 100; i++) {
+    /// 	let i = undefined;
+    /// }
+    /// ```
+    /// ```js,expect_diagnostic
+    /// let f = /**/undefined/**/ ;
+    /// ```
+    /// ### Valid
+    ///
+    /// ```js
+    /// var a = 1;
+    /// ```
+    /// ```js
+    /// class Foo {
+    /// 	bar = undefined;
+    /// }
+    /// ```
+    ///
+    pub NoUselessUndefinedInitialization {
+        version: "next",
+        name: "noUselessUndefinedInitialization",
+        sources: &[RuleSource::Eslint("no-undef-init")],
+        fix_kind: FixKind::Unsafe,
+        recommended: false,
+    }
+}
+
+impl Rule for NoUselessUndefinedInitialization {
+    type Query = Ast<JsVariableDeclaration>;
+    type State = (String, TextRange);
+    type Signals = Vec<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let let_or_var_kind = node.is_let() || node.is_var();
+
+        let mut signals = vec![];
+
+        if !let_or_var_kind {
+            return signals;
+        }
+
+        for declarator in node.declarators() {
+            let Ok(decl) = declarator else { continue };
+
+            let Some(keyword) = decl
+                .initializer()
+                .map(|initializer| initializer.expression())
+                .and_then(|expression| expression.ok())
+                .and_then(|expression| expression.as_js_reference_identifier())
+            else {
+                continue;
+            };
+
+            if keyword.is_undefined() {
+                let decl_range = decl.range();
+                let Some(binding_name) = decl.id().ok().map(|id| id.text()) else {
+                    continue;
+                };
+                signals.push((binding_name, decl_range));
+            }
+        }
+
+        signals
+    }
+
+    fn diagnostic(_ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        Some(RuleDiagnostic::new(
+            rule_category!(),
+            state.1,
+            markup! {
+                "It's not necessary to initialize "<Emphasis>{state.0}</Emphasis>" to undefined."
+            }).note("A variable that is declared and not initialized to any value automatically gets the value of undefined.")
+        )
+    }
+
+    fn action(ctx: &RuleContext<Self>, state: &Self::State) -> Option<JsRuleAction> {
+        let declarators = ctx.query().declarators();
+
+        let initializer = declarators
+            .clone()
+            .into_iter()
+            .find(|el| {
+                el.as_ref()
+                    .ok()
+                    .and_then(|element| element.id().ok())
+                    .is_some_and(|id| id.text() == state.0)
+            })
+            .map(|decl| decl.ok())?
+            .and_then(|declarator| declarator.initializer())?;
+
+        let mut mutation = declarators.begin();
```

Comment:
You shouldn't create a mutation this way.

```rs
let mut mutation = ctx.root().begin();
```

Can't remember the exact syntax. I suggest you look at other rules